### PR TITLE
Fix Ubuntu 20.04 runtime errors

### PR DESCRIPTION
**Pull request by ikurek**
1. Update `libgit2` to version [1.5.0](https://github.com/libgit2/libgit2/releases/tag/v1.5.0). This should solve #11, because an upstream fix was added to dynamically load OpenSSL libraries.
2. Updated `git2go` to version `v35`. It was required to support `libgit2` in the latest version
3. Bitrise step dependencies were updated to stable versions
4. Bitrise Ubuntu stack was changed to 20.04. Version 16.04 has reached end of life last year
5.  Fixed an error in input documentation. The `override_field` value is handled like this:

```go
if stepConfig.Overwrite {
    jiraWorker.UpdateBuildForIssues(issueKeys, build)
} else {
    jiraWorker.UpdateBuildForIssuesMultiField(issueKeys, build)
}
```

So when it's set to `true` the value gets overriden, but the docs stated otherwise.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM bitriseio/docker-bitrise-base:latest
```
Comment (fredyshox) [inline Dockerfile]: Building a custom image is not necessary, as you can run it interactively in the container while preserving current state of source files:
```sh
docker run -v $(pwd):/bitrise/src bitriseio/docker-bitrise-base:latest /bin/bash -c "bitrise update && bitrise run test"
```

I suggest replacing this file with the script above, called `test-linux.sh`

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1 @@
+vendor/**/* linguist-generated=true
```
Comment (fredyshox) [inline .gitattributes]: `vendor/` files still show up in diff stats 🤔

File: Dockerfile (hunk 1 above)
Comment (ikurek) [reply Dockerfile]: Done, I've removed the Dockerfile and replaced it with provided bash script

File: .gitattributes (hunk 2 above)
Comment (ikurek) [reply .gitattributes]: I've updated the `.gitattributes` file with

```
vendor/** linguist-vendored
go.sum linguist-generated
```

But the files still seem to be visible 🤔

Comment (fredyshox) [reply .gitattributes]: Let's leave it as is.

File: Dockerfile (hunk 1 above)
Comment (fredyshox) [reply Dockerfile]: cool 👍

**Review by fredyshox — APPROVED** [review]
LGTM!

**Comment by YaminYazdanpanah** [comment]
Hi,
Despite using the latest version, I am still experiencing an issue.

```
+++ dirname /tmp/bitrise829924854/step_src/step.sh
++ cd /tmp/bitrise829924854/step_src
++ pwd
+ THIS_SCRIPT_DIR=/tmp/bitrise829924854/step_src
++ uname -s
+ '[' Linux == Linux ']'
+ DEB_FILES=("${THIS_SCRIPT_DIR}/ubuntu/products/"*.deb)
+ dpkg -i /tmp/bitrise829924854/step_src/ubuntu/products/libgit2_1.5.0-1_amd64.deb
Selecting previously unselected package libgit2.
(Reading database ... 93169 files and directories currently installed.)
Preparing to unpack .../libgit2_1.5.0-1_amd64.deb ...
Unpacking libgit2 (1.5.0) ...
Setting up libgit2 (1.5.0) ...
+ ldconfig
++ mktemp -d
+ TMP_GOPATH_DIR=/tmp/tmp.1lzTrQ8E59
+ GO_PACKAGE_NAME=github.com/Holdapp/bitrise-step-jira-build
+ FULL_PACKAGE_PATH=/tmp/tmp.1lzTrQ8E59/src/github.com/Holdapp/bitrise-step-jira-build
+ mkdir -p /tmp/tmp.1lzTrQ8E59/src/github.com/Holdapp/bitrise-step-jira-build
+ rsync -avh --quiet /tmp/bitrise829924854/step_src/ /tmp/tmp.1lzTrQ8E59/src/github.com/Holdapp/bitrise-step-jira-build/
+ export GOPATH=/tmp/tmp.1lzTrQ8E59
+ GOPATH=/tmp/tmp.1lzTrQ8E59
+ go run /tmp/tmp.1lzTrQ8E59/src/github.com/Holdapp/bitrise-step-jira-build/main.go
# github.com/Holdapp/bitrise-step-jira-build/vendor/github.com/libgit2/git2go/v34
/usr/bin/ld: warning: libssl.so.1.0.0, needed by /usr/local/lib/libgit2.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libcrypto.so.1.0.0, needed by /usr/local/lib/libgit2.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `ERR_error_string@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_write@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `X509_get_subject_name@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `ERR_lib_error_string@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_CTX_ctrl@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `sk_num@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_read@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `X509_NAME_get_entry@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `X509_free@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_set_bio@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SHA256_Update@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_get_error@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SHA256_Init@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `CRYPTO_free@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `GENERAL_NAMES_free@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_new@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `HMAC_Update@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_CTX_set_verify@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_get_verify_result@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `CRYPTO_num_locks@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_CTX_set_default_verify_paths@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `ASN1_STRING_to_UTF8@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `ERR_get_error@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `HMAC_Final@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `i2d_X509@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `CRYPTO_THREADID_set_callback@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `DES_set_key@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `DES_ecb_encrypt@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `CRYPTO_THREADID_set_numeric@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSLv23_method@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_shutdown@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `ASN1_STRING_data@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_free@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `ASN1_STRING_length@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_ctrl@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_library_init@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `HMAC_CTX_cleanup@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_connect@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_CTX_new@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `sk_value@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `HMAC_Init_ex@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `X509_get_ext_d2i@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_CTX_free@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `CRYPTO_malloc@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `X509_NAME_ENTRY_get_data@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `RAND_bytes@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_get_peer_certificate@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `MD4@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_CTX_load_verify_locations@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `X509_NAME_get_index_by_NID@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `ERR_error_string_n@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SHA256_Final@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `CRYPTO_set_locking_callback@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `ASN1_STRING_type@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_CTX_set_cipher_list@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `BIO_new@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `SSL_load_error_strings@OPENSSL_1.0.0'
/usr/bin/ld: /usr/local/lib/libgit2.so: undefined reference to `EVP_md5@OPENSSL_1.0.0'
collect2: error: ld returned 1 exit status
```

Any idea?

**Pull request merged**